Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -0,0 +1,263 @@
+                }
+                else if (targetOs.ContainsIgnoreCase("mac"))
+                {
+                    return OperatingSystem.MacOS;
+                }
+                else if (targetOs.ContainsIgnoreCase("tvos"))
+                {
+                    return OperatingSystem.TvOs;
+                }
+                else if (targetOs.ContainsIgnoreCase("linux"))
+                {
+                    return OperatingSystem.Linux;
+                }
+                else if (targetOs.ContainsIgnoreCase("win") ||
+                    targetOs.ContainsIgnoreCase("uap") ||
+                    targetOs.ContainsIgnoreCase("win") ||
+                    targetOs == ".NETFramework")
+                {
+                    return OperatingSystem.Windows;
+                }
+                else
+                {
+                    return $"{frameworkName.Identifier} is an unknown target os";
+                }
+            });
+
+        public static string SHA256Hash(byte[] bytes)
+        {
+            using (var sha256 = System.Security.Cryptography.SHA256.Create())
+            {
+                return BitConverter.ToString(sha256.ComputeHash(bytes));
+            }
+        }
+
+        public static string GetHostCpuArchitecture =>
+            WrapInTryCatch(() => ConvertArchitectureToMetricsVersion(RuntimeInformation.OSArchitecture.ToString()));
+
+        public static string GetTargetCpuArchitecture(ModuleDefinition module) =>
+            WrapInTryCatch(() => ConvertArchitectureToMetricsVersion(module.Architecture.ToString()));
+
+        public static byte[] GenerateComputerIdentifier =>
+            NetworkInterface.GetAllNetworkInterfaces()
+                .Where(n => n.Name == "en0" || (n.OperationalStatus == OperationalStatus.Up && n.NetworkInterfaceType != NetworkInterfaceType.Loopback))
+                .Select(n => n.GetPhysicalAddress().GetAddressBytes())
+                .FirstOrDefault();
+
+        public static string ConvertPlatformIdOsToMetricVersion(PlatformID platformID) =>
+            WrapInTryCatch(() =>
+                platformID switch
+                {
+                    PlatformID.Win32NT or PlatformID.Win32S or
+                    PlatformID.Win32Windows or PlatformID.WinCE => OperatingSystem.Windows,
+                    PlatformID.MacOSX => OperatingSystem.MacOS,
+                    PlatformID.Unix => OperatingSystem.Linux,
+                    _ => $"{platformID} is an unsupported operating system."
+                });
+
+        public static (string Name, string Version) GetFrameworkAndVersion(ModuleDefinition module, Config config)
+        {
+            if (config.TargetFramework.ContainsIgnoreCase("unity"))
+            {
+                return config.TargetFramework.ContainsIgnoreCase("editor") ?
+                    (Framework.UnityEditor, config.TargetFrameworkVersion) :
+                    (Framework.Unity, config.TargetFrameworkVersion);
```

Comment:
What's the difference between unity and unityEditor?

Reply:
The editor is when you're running the project in the editor and the player is when you're running the binary of your game as standalone.
So we need to know if it's weaving to target the editor or to target the standalone binary.